### PR TITLE
Create sample-by-pathway dataset

### PR DESCRIPTION
**Pull request by stephenshank**
First attempt, would be very grateful for some code review!

### CODE REVIEW

**Comment by stephenshank** [comment]
I may have done something undesirable with git/github... for some reason, changes from my last pull request are showing up in this one. The file to look at is `5.hetnet-pathways.ipynb`.

**Comment by dhimmel** [comment]
> I may have done something undesirable with git/github...

You need to rebase your pull request on top of the latest upstream master. This will be a little more difficult because you based your pull request off of your master branch. Usually you make pull requests using a new branch, so your master branch can easily stay synced with upstream.

If you've [configured `cognoma/cancer-data`](https://help.github.com/articles/configuring-a-remote-for-a-fork/) to be upstream, then you should be able to do the following, while your master branch is checked out:

```
git fetch upstream
git rebase upstream/master
git push --force
```

File: scripts/5.hetnet-pathways.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  gene.identifier as gene_id, 
+  pathway.name as pathway_name,
+  pathway.identifier as pathway_id
+'''
+
+driver = GraphDatabase.driver("bolt://neo4j.het.io")
+hetnet_results = pd.DataFrame()
+with driver.session() as session:
+    result = session.run(query)
+    hetnet_results = ( pd.DataFrame((x.values() for x in result), columns=result.keys())
```
Comment (dhimmel) [inline scripts/5.hetnet-pathways.py:37]: Can we call `hetnet_results` something more informative such as `pathway_df`?

Comment (dhimmel) [reply scripts/5.hetnet-pathways.py:37]: not the prettiest spacing

File: scripts/5.hetnet-pathways.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# In[8]:
+
+number_of_samples = len(mutation_df)
+number_of_pathways = len(pathways) 
+sample_pathway_df = pd.DataFrame(np.zeros((number_of_samples, number_of_pathways), dtype=np.int),
+                                 index=mutation_df.index,
+                                 columns=pathways)
+
+
+# Now populate this data frame. This is a slow Python loop, hence the progress bar. It takes a few minutes on my laptop. The idea is to loop over all gene-pathway interactions in the hetnet query. If the gene is in the Cognoma dataset, we grab the pathway id in that gene-pathway interaction. We look at Cognoma samples where that gene is labeled 1, i.e., at Cognoma samples that have a mutation in that gene, and grab the corresponding indices. Then, in the pathway matrix all samples get the associated pathway tagged as a 1, since they have a mutated gene that participates in that pathway.
```
Comment (dhimmel) [inline scripts/5.hetnet-pathways.py:99]: I would consider the following strategy:
1. melt mutation_df so you get at `sample_id, entrez_gene_id, mutation` dataframe.
2. filter that for `mutation == 1`
3. merge with `pathway_df`
4. pivot_table to create a sample by pathway matrix. See [this example](https://github.com/cognoma/cancer-data/blob/54140cf6addc48260c9723213c40b628d7c861da/scripts/2.TCGA-process.py#L157-L161) for this step.

Hopefully that's a little bit faster and more readable. You won't be able to use the progress bar anymore.

File: scripts/5.hetnet-pathways.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+# 
+# Imports...
+
+# In[1]:
+
+from neo4j.v1 import GraphDatabase
+import pandas as pd
+import numpy as np
+import os
+from ipywidgets import FloatProgress
```
Comment (dhimmel) [inline scripts/5.hetnet-pathways.py:16]: Didn't know about this... looks cool and nice that it's builtin to notebooks.

File: scripts/5.hetnet-pathways.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+
+# Initialize a samples by pathways data frame...
+
+# In[8]:
+
+number_of_samples = len(mutation_df)
+number_of_pathways = len(pathways) 
+sample_pathway_df = pd.DataFrame(np.zeros((number_of_samples, number_of_pathways), dtype=np.int),
+                                 index=mutation_df.index,
```
Comment (dhimmel) [inline scripts/5.hetnet-pathways.py:95]: I like only a single indent here. May be personal preference but this style has is a space hog. Maybe move `np.zeros...` to a newline to improve readability.

**Review by dhimmel — CHANGES_REQUESTED** [review]
Very nice. Awesome to see that you were able to interact programmatically with Hetionet. I made a few comments in addition to the rebase required as discussed above.

How long did the queries take? May also be nice to do this for Biological Processes.

Also I think we may want to export a `pathway-info.tsv` with identifier, name, n_mutations, and maybe some of the other Hetionet attributes such as source.

**Pull request closed without merging**